src/it/scala/djinni/GeneratorIntegrationTest.scala

```diff
@@ -78,5 +79,33 @@ class GeneratorIntegrationTest extends IntegrationTest with GivenWhenThen {
         assertFileContentEquals(idlFile, OBJCPP, objcppFilenames)
       }
     }
+
+    it("should be able to only generate Java output") {
+      val outputPath = "src/it/resources/result/only_java_out"
+      When("calling the generator with just `--java-out`")
+      val output: String = djinni(s"--idl src/it/resources/all_datatypes.djinni --java-out $outputPath")
+      Then("the generator should successfully generate just java output")
+      output should equal ("Parsing...\nResolving...\nGenerating...\n")
+      assertFileExists(s"$outputPath/AllDatatypes.java")
+    }
+
+    it("should be able to only generate Obj-C output") {
+      val outputPath = "src/it/resources/result/only_objc_out"
+      When("calling the generator with just `--objc-out`")
+      val output = djinni(s"--idl src/it/resources/all_datatypes.djinni --objc-out $outputPath")
+      Then("the generator should successfully generate just objc output")
+      output should equal ("Parsing...\nResolving...\nGenerating...\n")
+      assertFileExists(s"$outputPath/AllDatatypes.h")
+      assertFileExists(s"$outputPath/AllDatatypes.mm")
```

Comment:
Is it possible to add some assert that file not exist for Java?
just to be sure that this is not generated

Reply:
mh. good idea 👍

Reply:
hm, now that I think about it, this test would not make a lot of sense:

If no `--java-out` parameter is given, then the generator does not know where to put the java files, even if he erroneously decided to create them. As a tester I don't know where to expect these files, so I cannot test for their absense. 

If I wanted to be sure that the generator did not generate Java-files in the objc-out directory, I can already rely on the [tests above](https://github.com/cross-language-cpp/djinni-generator/blob/main/src/it/scala/djinni/GeneratorIntegrationTest.scala#L59). They make sure that all files are written to where they should be.

Reply:
agreed, 
 `find ./ -name "*.java"` , and with h/mm for the java case,  in the current working directory and output path, might give no 100% security nothing has been written (without creating a chroot environment) but it might be a nice placebo to have.
Question is if it's too much hassle to create .... and it's also not super important
So I approve and let the decision to you :-)